client/app/components/items-list/classes/Sorter.js

```diff
@@ -0,0 +1,71 @@
+import { isString, sortBy } from 'lodash';
+
+const ORDER_BY_REVERSE = '-';
+
+export function compile(field, reverse) {
+  if (!field) {
+    return null;
+  }
+  return reverse ? ORDER_BY_REVERSE + field : field;
+}
+
+export function parse(compiled) {
+  compiled = isString(compiled) ? compiled : '';
+  const reverse = compiled.startsWith(ORDER_BY_REVERSE);
+  if (reverse) {
+    compiled = compiled.substring(1);
+  }
+  const field = compiled !== '' ? compiled : null;
+  return { field, reverse };
+}
+
+export default class Sorter {
+  field = null;
+
+  reverse = false;
+
+  get compiled() {
+    return compile(this.field, this.reverse);
+  }
+
+  set compiled(value) {
+    const { field, reverse } = parse(value);
+    this.field = field;
+    this.reverse = reverse;
+  }
+
+  setField(value) {
+    this.field = isString(value) && (value !== '') ? value : null;
+  }
+
+  setReverse(value) {
+    this.reverse = !!value; // cast to boolean
+  }
+
+  constructor({ orderByField, orderByReverse } = {}) {
+    this.setField(orderByField);
+    this.setReverse(orderByReverse);
+  }
+
+  toggleField(field) {
+    if (!isString(field) || (field === '')) {
```

Comment:
TS, WE AWAIT THEE!

Reply:
we'll anyway need this check (at least `field === ''), even with TS

Reply:
Cause it's being taken from the url I suppose?

Reply:
Cause it comes from "outside", and TS will check only data type; but for this class field cannot be empty string as well (does not make sense). Probably it's not a big deal to omit this check at all with TS, but at least now we need it.